From 71a6d6fb3768c4668c7acb2dcb62ee8260d109c7 Mon Sep 17 00:00:00 2001
From: Jernej Skrabec <jernej.skrabec@gmail.com>
Date: Fri, 22 May 2026 16:39:26 +0000
Subject: [PATCH 26/33] drm/sun4i: mixer: Add A523 mixer compatible

Allwinner A523/A527/T527 Display Engine 3.5 shares the DE 3.3 mixer
register layout, so reuse the SUN8I_MIXER_DE33 type for the new
compatible.

Signed-off-by: Jernej Skrabec <jernej.skrabec@gmail.com>
---
 drivers/gpu/drm/sun4i/sun8i_mixer.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/gpu/drm/sun4i/sun8i_mixer.c b/drivers/gpu/drm/sun4i/sun8i_mixer.c
index 7cd72dd6f..61812f8d5 100644
--- a/drivers/gpu/drm/sun4i/sun8i_mixer.c
+++ b/drivers/gpu/drm/sun4i/sun8i_mixer.c
@@ -914,6 +914,11 @@ static const struct sun8i_mixer_cfg sun50i_h616_mixer0_cfg = {
 	.mod_rate	= 600000000,
 };
 
+static const struct sun8i_mixer_cfg sun55i_a523_mixer0_cfg = {
+	.de_type	= SUN8I_MIXER_DE33,
+	.mod_rate	= 600000000,
+};
+
 static const struct of_device_id sun8i_mixer_of_table[] = {
 	{
 		.compatible = "allwinner,sun8i-a83t-de2-mixer-0",
@@ -963,6 +968,10 @@ static const struct of_device_id sun8i_mixer_of_table[] = {
 		.compatible = "allwinner,sun50i-h616-de33-mixer-0",
 		.data = &sun50i_h616_mixer0_cfg,
 	},
+	{
+		.compatible = "allwinner,sun55i-a523-de35-mixer-0",
+		.data = &sun55i_a523_mixer0_cfg,
+	},
 	{ }
 };
 MODULE_DEVICE_TABLE(of, sun8i_mixer_of_table);
-- 
2.53.0

